260 submit component

james-perretta opened the pull request:
Fixes #260 
Fixes #265

james-perretta committed:
Temp, working on files to submit processing

james-perretta committed:
Implemented Submit component.
Also added progress bar component.

james-perretta committed:
Added snapshot tests for progress bar.

james-perretta committed:
Wrote tests for submit.vue.
Also added logic to the submit button text so that it tells the user when their submission will (based on soft closing time) use a late day.

james-perretta committed:
Fixed linter errors.

james-perretta committed:
Added stub of User.get_num_late_days to test_project_views.ts

ashberg commented on src/components/file_upload.vue:
```diff
@@ -210,6 +210,10 @@ export default class FileUpload extends Vue {
   width: 100%;
+
+  * {
+    pointer-events: none;
```
The "Upload from your computer" button is no longer clickable.

james-perretta replied:
Oops

ashberg commented on src/components/project_view/submit.vue:
```diff
@@ -0,0 +1,472 @@
+        return true;
+      }
+    }
+
```
There's an extra space here.

james-perretta replied:
Blank line or not before a return statement isn't something that necessarily needs to be standardized, but sure, I can take out this one.

james-perretta committed:
Fixed handling of dragenter and dragleave events in file_upload.vue.
Implementation now uses a counter so that dragging over the 'upload from your computer' button doesn't turn off the highlighting.
This implementation also makes sure that the button is still clickable.

james-perretta committed:
Fixed unsupported format warnings from moment.

james-perretta committed:
wat

james-perretta committed:
Refactored modals in submit.vue to use new interface.
